File: res/jburkardt/people.sc.fsu.edu/~jburkardt/m_src/euler_test/euler_test.m
function euler_test ( )

%*****************************************************************************80
%
%% euler_test tests euler.
%
%  Licensing:
%
%    This code is distributed under the GNU LGPL license.
%
%  Modified:
%
%    25 April 2020
%
%  Author:
%
%    John Burkardt
%
  addpath ( '../euler' );

  timestamp ( );
  fprintf ( 1, '\n' );
  fprintf ( 1, 'euler_test\n' );
  fprintf ( 1, '  MATLAB/Octave version %s\n', version ( ) );
  fprintf ( 1, '  Test euler().\n' );

  n = 50;
  humps_euler ( n );
%
%  Terminate.
%
  fprintf ( 1, '\n' );
  fprintf ( 1, 'euler_test:\n' );
  fprintf ( 1, '  Normal end of execution.\n' );
  fprintf ( 1, '\n' );
  timestamp ( );

  rmpath ( '../euler' );

  return
end
function timestamp ( )

%*****************************************************************************80
%
%% timestamp prints the current YMDHMS date as a timestamp.
%
%  Licensing:
%
%    This code is distributed under the GNU LGPL license.
%
%  Modified:
%
%    14 February 2003
%
%  Author:
%
%    John Burkardt
%
  t = now;
  c = datevec ( t );
  s = datestr ( c, 0 );
  fprintf ( 1, '%s\n', s );

  return
end

File: res/jburkardt/people.sc.fsu.edu/~jburkardt/r_src/euler_test/euler_test.R
#*****************************************************************************80
#
## euler_test tests euler.
#
#  Licensing:
#
#    Copyright 2016 James P. Howard, II
#
#    The computer code and data files on this web page are distributed under
#    https://opensource.org/licenses/BSD-2-Clause, the BSD-2-Clause license.
#
#  Modified:
#
#    23 March 2020
#
#  Author:
#
#    John Burkardt
#
cat ( date ( ), "\n" )
cat ( "\n" )
cat ( "euler_test\n" )
cat ( " ", version$version.string, "\n" )
cat ( "  euler uses the Euler method to solve an ordinary differential equation.\n" )

source ( "/home/burkardt/public_html/r_src/euler/euler.R" )  

f <- function ( x, y ) { y / ( 2.0 * x + 1.0 ) }
cat ( "\n" )
cat ( "  f(x,y) = y/(2*x+1)\n" )
cat ( "\n" )

a = 0.0
b = 1.0
h = 0.2
n = 5
solution <- euler ( f, a, b, h, n )
print ( solution )
#
#  Terminate.
#
cat ( "\n" )
cat ( "euler_test\n" )
cat ( "  Normal end of execution.\n" )
cat ( date ( ), "\n" )

quit ( )
